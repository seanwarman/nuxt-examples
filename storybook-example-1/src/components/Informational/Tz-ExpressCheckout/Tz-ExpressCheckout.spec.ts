/**
 * @vitest-environment jsdom
 */

import { VueWrapper, mount } from '@vue/test-utils';
import { expect, describe, beforeEach, it } from 'vitest';
import TzExpressCheckout from './Tz-ExpressCheckout.vue';

let wrapper: VueWrapper;

describe('TzExpressCheckout', () => {
  beforeEach(() => {
    wrapper = mount(TzExpressCheckout, {
      props: {
        options: [
          {
            heading: 'x',
            subHeading: 'y',
            value: 'z'
          },
          {
            heading: 'a',
            subHeading: 'b',
            value: 'c'
          }
        ]
      }
    });
  });

  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  describe('when an option is clicked', () => {
    it("will emit a 'option:click' event with the appropriate value", () => {
      wrapper.find('button:last-of-type').trigger('click');
      expect(wrapper.emitted('option:click')).toBeTruthy();
      expect(wrapper.emitted('option:click')![0]).toEqual(['c']);
    });
  });
});
