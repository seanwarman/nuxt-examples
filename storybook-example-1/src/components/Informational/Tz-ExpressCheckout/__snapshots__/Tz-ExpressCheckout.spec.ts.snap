// Vitest Snapshot v1

exports[`TzExpressCheckout > renders correctly 1`] = `
<section
  class="flex flex-col items-center gap-5 text-gray-800 sm:gap-16"
>
  <div
    class="flex flex-col gap-5 sm:flex-row"
  >
    
    <button
      class="group block min-h-[114px] w-[250px] rounded-sm border p-4 px-5 text-center sm:py-7"
    >
      <span
        class="block pb-3 text-base font-black uppercase leading-5 tracking-wider group-hover:underline group-focus:underline"
      >
        x
      </span>
      <span
        class="mb-4 block min-h-[32px] text-sm leading-4"
      >
        y
      </span>
    </button>
    <button
      class="group block min-h-[114px] w-[250px] rounded-sm border p-4 px-5 text-center sm:py-7"
    >
      <span
        class="block pb-3 text-base font-black uppercase leading-5 tracking-wider group-hover:underline group-focus:underline"
      >
        a
      </span>
      <span
        class="mb-4 block min-h-[32px] text-sm leading-4"
      >
        b
      </span>
    </button>
    
  </div>
  <!--v-if-->
</section>
`;
